rust/src/composepost.rs

```diff
@@ -788,29 +788,46 @@ fn convert_path_to_tmpfiles_d_recurse(
 ///
 /// For example:
 ///  - Symlink /usr/local -> /var/usrlocal
-///  - Symlink /var/lib/alternatives -> /usr/lib/alternatives
-///  - Symlink /var/lib/vagrant -> /usr/lib/vagrant
+///  - If present, symlink /var/lib/alternatives -> /usr/lib/alternatives
+///  - If present, symlink /var/lib/vagrant -> /usr/lib/vagrant
 #[context("Preparing symlinks in rootfs")]
 pub fn rootfs_prepare_links(rootfs_dfd: i32) -> CxxResult<()> {
     let rootfs = crate::ffiutil::ffi_view_openat_dir(rootfs_dfd);
 
+    // Unconditionally drop /usr/local and replace it with a symlink.
     rootfs
         .remove_all("usr/local")
         .context("Removing /usr/local")?;
-    let state_paths = &["usr/lib/alternatives", "usr/lib/vagrant"];
-    for entry in state_paths {
+    ensure_symlink(&rootfs, "../var/usrlocal", "usr/local")
+        .context("Creating /usr/local symlink")?;
+
+    // Move existing content to /usr/lib, then put a symlink in its
+    // place under /var/lib.
+    rootfs
+        .ensure_dir_all("usr/lib", 0o0755)
+        .context("Creating /usr/lib")?;
+    let content_dirs = &["alternatives", "vagrant"];
+    for entry in content_dirs {
+        let varlib_path = format!("var/lib/{}", entry);
+        let is_var_dir = rootfs
+            .metadata_optional(&varlib_path)?
+            .map(|m| m.is_dir())
+            .unwrap_or(false);
```

Comment:
(Random aside, this use of combinators to me is a great micro-example of Rust's elegance)